apps/web/src/views/universalFarms/components/StakeModal.tsx

```diff
@@ -0,0 +1,111 @@
+import styled from 'styled-components'
+import { AtomBox, AutoColumn, Button, Image, Modal, ModalV2, StyledTooltip, Text } from '@pancakeswap/uikit'
+import { useTranslation } from '@pancakeswap/localization'
+import { useTheme } from '@pancakeswap/hooks'
+import { LightCard } from 'components/Card'
+
+const Arrow = styled.div`
+  position: absolute;
+  top: 0px;
+  transform: translate3d(0px, 62px, 0px);
+  right: 4px;
+  &::before {
+    content: '';
+    transform: rotate(45deg);
+    background: var(--colors-backgroundAlt);
+    position: absolute;
+    width: 10px;
+    height: 10px;
+    border-radius: 2px;
+    z-index: -1;
+  }
+`
+
+interface IStakeModalProps {
+  outOfRangeUnstaked: boolean
+  onStake?: (e: React.MouseEvent) => void
+  onUnStake?: (e: React.MouseEvent) => void
+  onDismiss: () => void
+  isOpen: boolean
+}
+
+export const StakeModal: React.FC<React.PropsWithChildren<IStakeModalProps>> = ({
+  outOfRangeUnstaked,
+  onStake,
+  onUnStake,
+  isOpen,
+  onDismiss,
+  children,
+}) => {
+  const { t } = useTranslation()
+  const { isDark } = useTheme()
+  return (
+    <ModalV2 isOpen={isOpen} onDismiss={onDismiss} closeOnOverlayClick>
+      <Modal
+        title={outOfRangeUnstaked ? t('Staking') : t('Unstaking')}
```

Comment:
should `outOfRange` as a condition?

I think just `staked => Unstaking`, `!staked => Staking`

Reply:
nope, this is a case of click a stake button for a inactive position, so this case must be !staked.